ui/lib/kv/addon/components/kv-secret-form.hbs

```diff
@@ -6,24 +6,41 @@
   </ToolbarFilters>
 </Toolbar>
 
+{{#if this.showAlert}}
+  <Hds::Alert data-test-secret-version-alert @type="inline" @color="warning" class="has-top-bottom-margin" as |A|>
+    <A.Title>Warning</A.Title>
+    <A.Description>
+      You are creating a new version based on data from Version
+      {{@previousVersion}}. The current version for
+      <code>{{@secret.path}}</code>
+      is Version
+      {{@metadata.currentVersion}}.
+    </A.Description>
+  </Hds::Alert>
+{{/if}}
+
 <form {{on "submit" (perform this.save)}}>
   <div class="box is-sideless is-fullwidth is-bottomless">
     <NamespaceReminder @mode="create" @noun="secret" />
     <MessageError @model={{@secret}} @errorMessage={{this.errorMessage}} />
 
     {{#let (find-by "name" "path" @secret.allFields) as |attr|}}
-      <FormField
-        @attr={{attr}}
-        @model={{@secret}}
-        @modelValidations={{this.modelValidations}}
-        @onKeyUp={{this.pathValidations}}
-      />
+      {{#if @previousVersion}}
+        <ReadonlyFormField @attr={{attr}} @value={{get @secret attr.name}} />
+      {{else}}
+        <FormField
+          @attr={{attr}}
+          @model={{@secret}}
+          @modelValidations={{this.modelValidations}}
+          @onKeyUp={{this.pathValidations}}
```

Comment:
Curious what `this.pathValidations` is for? I'm not seeing a reference to it in the component.

Reply:
It's an action in the form 
```js
 @action
  pathValidations() {
    // check path attribute warnings on key up
    const { state } = this.args.secret.validate();
    if (state?.path?.warnings) {
      // only set model validations if warnings exist
      this.modelValidations = state;
    }
  }

Reply:
In `kv-secret-form.js`? Unless I'm crazy I'm not seeing it. I didn't pull down the branch though so maybe GH is playing some trickery on me 😁

Reply:
Haha, yep! Did you expand the full file? Github decided a weird place to truncate the file...
<img width="868" alt="Screenshot 2023-08-03 at 2 23 27 PM" src="https://github.com/hashicorp/vault/assets/68122737/038f75f6-c173-4f61-9e7d-2777c40ffb55">

Reply:
I did I swear!! Homer fading backwards through the hedge emoji